packages/phoenix-client/src/phoenix/client/helpers/sdk/openai/chat.py

```diff
@@ -570,7 +572,31 @@ def _from_tool_choice(
             function_name=function["name"],
         )
         return choice_function_tool
-    assert_never(obj["type"])
+    if obj.get("type") == "allowed_tools":
+        raw_values: Any = obj.get("values")
+        if isinstance(raw_values, (list, tuple)):
+            names: list[str] = []
+            seq = cast(Sequence[Mapping[str, Any]], raw_values)
+            for value in seq:
+                if value.get("type") == "function":
+                    fn_any: Any = value.get("function")
+                    if isinstance(fn_any, Mapping):
+                        fn = cast(Mapping[str, Any], fn_any)
+                        name_any: Any = fn.get("name")
+                        if isinstance(name_any, str):
+                            names.append(name_any)
+            unique = list(dict.fromkeys(names))
+            if len(unique) == 1:
+                # Return the specific function tool if there is only one function
+                return v1.PromptToolChoiceSpecificFunctionTool(
+                    type="specific_function",
+                    function_name=unique[0],
+                )
+        # Otherwise, degrade to zero_or_more
+        return v1.PromptToolChoiceZeroOrMore(type="zero_or_more")
+    if obj.get("type") == "custom":
+        return v1.PromptToolChoiceZeroOrMore(type="zero_or_more")
+    assert_never(cast(Never, obj["type"]))
```

Comment:
Since the current type-checking approach helped us catch upstream changes, is there a reason we don’t want to continue with it?
```suggestion
    assert_never(obj["type"])
```

Reply:
oh oops, this wasn't intended to be changed